pkg/models/batch_transformation.go

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2020-present Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This software is made available by Snowplow Analytics, Ltd.,
+ * under the terms of the Snowplow Limited Use License Agreement, Version 1.0
+ * located at https://docs.snowplow.io/limited-use-license-1.0
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING OR DISTRIBUTING ANY PORTION
+ * OF THE SOFTWARE, YOU AGREE TO THE TERMS OF SUCH LICENSE AGREEMENT.
+ */
+
+package models
+
+import "time"
+
+// MessageBatch houses batches of messages, for batch transformations to operate across
+type MessageBatch struct {
+	OriginalMessages    []*Message        // Most targets will use the data from here, but where we have a http templating transformation, we would use this to ack batches of messages
```

Comment:
I assume you need to pass this around so you can still hit the `ack` on each individual message that has been batched up right?

Reply:
Yes, but also for the non-http targets, we use it to preserve the original behaviour.